gallery/src/styles/NxFormLayout/NxFormHorizontalLayoutExample.tsx

```diff
@@ -8,49 +8,46 @@ import React from 'react';
+  function validator(val: string) {
+    return val.length ? null : 'Must be non-empty';
+  }
+
   return (
     <form className="nx-form">
       <div className="nx-form-row">
         <div className="nx-form-group">
           <label className="nx-label">
-            <span className="nx-label__text">Horizontal layout</span>
-            <span className="nx-sub-label">
-              <NxFontAwesomeIcon icon={faCalendar}/>
-              <span>Sub-label with icon.</span>
-            </span>
-            <input type="text" className="nx-text-input"/>
+            <span className="nx-label__text">Label</span>
+            <NxStatefulTextInput/>
           </label>
         </div>
         <div className="nx-form-group">
-          <label className="nx-label">
-            <span className="nx-label__text">Second column</span>
-            <span className="nx-sub-label">This is a sub-label.</span>
-            <input type="text" className="nx-text-input"/>
+          <label className="nx-label nx-label--optional">
+            <span className="nx-label__text">Label</span>
+            <NxStatefulTextInput validator={validator}/>
           </label>
         </div>
-        <div className="nx-form-group">
-          <button className="nx-btn" type="button">Button</button>
-        </div>
-      </div>
-      <div className="nx-form-row">
-        <fieldset className="nx-fieldset">
-          <legend className="nx-label nx-label--optional">Checkboxes</legend>
-          <NxCheckbox isChecked={false}>Checkbox 1</NxCheckbox>
-          <NxCheckbox isChecked={true}>Checkbox 2</NxCheckbox>
-          <NxCheckbox isChecked={false}>Checkbox 3</NxCheckbox>
-        </fieldset>
-        <fieldset className="nx-fieldset">
-          <legend className="nx-label">Radio buttons</legend>
-          <NxRadio name="demo1" value="demo1" isChecked={true}>Radio Button 1</NxRadio>
-          <NxRadio name="demo2" value="demo2" isChecked={false}>Radio Button 2</NxRadio>
-          <NxRadio name="demo3" value="demo3" isChecked={true}>Radio Button 3</NxRadio>
-        </fieldset>
       </div>
+      <fieldset className="nx-fieldset">
+        <legend className="nx-legend nx-legend--optional"><span className="nx-legend__text">Checkboxes</span></legend>
+        <NxCheckbox isChecked={false}>Checkbox 1</NxCheckbox>
+        <NxCheckbox isChecked={true}>Checkbox 2</NxCheckbox>
+        <NxCheckbox isChecked={false}>Checkbox 3</NxCheckbox>
+      </fieldset>
+      <fieldset className="nx-fieldset">
+        <legend className="nx-legend"><span className="nx-legend__text">Radio buttons</span></legend>
+        <NxRadio name="demo1" value="demo1" isChecked={true}>Radio Button 1</NxRadio>
+        <NxRadio name="demo2" value="demo2" isChecked={false}>Radio Button 2</NxRadio>
+        <NxRadio name="demo3" value="demo3" isChecked={false}>Radio Button 3</NxRadio>
+      </fieldset>
+      <footer className="nx-form-footer">
+        <NxButton>Cancel</NxButton>
```

Comment:
This needs `type="button"` or `type="reset"`

Reply:
> This needs `type="button"` or `type="reset"`

Here and all other buttons: https://github.com/sonatype/sonatype-react-shared-components/pull/122/commits/f6ac79f95d8b4d44ac16dd8a24ee6888b8076a46